src/common.c

```diff
@@ -78,14 +78,30 @@ int fi_poll_fd(int fd, int timeout)
 	return ret == SOCKET_ERROR ? -ofi_sockerr() : ret;
 }
 
+/*
+ * Find the location of the most significant bit set. Value 1~64
+ * means such bit is found. Value 0 means no bit is set.
+ */
+static inline int msb(uint64_t x)
+{
+	int pos = 0;
+
+	while (x) {
+		x >>= 1;
+		pos++;
+	}
+
+	return pos;
+}
+
 uint64_t fi_tag_bits(uint64_t mem_tag_format)
 {
-	return UINT64_MAX >> (ffsll(htonll(mem_tag_format)) -1);
+	return UINT64_MAX >> (64 - msb(mem_tag_format));
 }
```

Comment:
It's actually taken me more than just a couple seconds to figure out what these functions are trying to do.  I wonder if we can't come up with better names.  Something like ofi_max_tag() seems like a better option for this call.

Reply:
I agree ofi_max_tag() would be a better name.